### ISSUE TITLE
Figure out possibility of tracking logger objects

### ISSUE THREAD

**Issue by kornicameister** [kornicameister/loguru-mypy#6]
Consider two examples

```python
from loguru import logger

def a_foo(a: int) -> int:
    return a

# case 1
logger.debug('Should work because not lazy=True {}', a_foo)
# case 2
logger.opt(lazy=True).debug('Should fail, because callable {} has args', a_foo)
```

They key difference between these two calls is having one logger defined with `lazy=True` and the other without it. Loggers that are lazy will in fact try to call a function supplied as *arg* or *kwarg* but will succeed only if function has no arguments whatsoever. In the same time loggers that are not lazy will simple print out i.e.  `str(a_foo)` of a function.

That said:
- **case 1** is **not* something `loguru-mypy` should flag
- **case 2** is something `loguru-mypy` should flag
